src/cargo/sources/registry/index.rs

```diff
@@ -742,7 +754,7 @@ impl Summaries {
 const CURRENT_CACHE_VERSION: u8 = 3;
 
 impl<'a> SummariesCache<'a> {
-    fn parse(data: &'a [u8], last_index_update: &str) -> CargoResult<SummariesCache<'a>> {
+    fn parse(data: &'a [u8]) -> CargoResult<SummariesCache<'a>> {
         // NB: keep this method in sync with `serialize` below
```

Comment:
Do we need to change `serialize` to match?

Reply:
No. This isn't changing the on-disk format. The `last_index_update` argument was only used to check if the cache was up to date.

Instead of comparing versions here, this function returns the `index_version` along with the cached data so that it can be compared later.